app/src/androidTest/java/com/github/wnder/HistoryActivityTest.java

```diff
@@ -1,29 +1,76 @@
-import androidx.test.ext.junit.runners.AndroidJUnit4;
 
+import com.github.wnder.networkService.NetworkInformation;
+import com.github.wnder.networkService.NetworkModule;
+import com.github.wnder.networkService.NetworkService;
+import com.github.wnder.picture.LocalPicture;
+import com.github.wnder.picture.LocalPictureDatabase;
+import com.github.wnder.picture.PicturesDatabase;
 import com.github.wnder.user.GlobalUser;
 import com.github.wnder.user.GuestUser;
+import com.github.wnder.user.User;
 
 import org.junit.After;
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.Test;
-import org.junit.runner.RunWith;
+import org.junit.rules.RuleChain;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.concurrent.CompletableFuture;
+
+import dagger.hilt.android.testing.BindValue;
+import dagger.hilt.android.testing.HiltAndroidRule;
+import dagger.hilt.android.testing.HiltAndroidTest;
+import dagger.hilt.android.testing.UninstallModules;
 
 import static androidx.test.espresso.Espresso.onView;
 import static androidx.test.espresso.action.ViewActions.click;
 import static androidx.test.espresso.assertion.ViewAssertions.matches;
-import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
 import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static java.lang.Thread.sleep;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
 
-@RunWith(AndroidJUnit4.class)
+@HiltAndroidTest
+@UninstallModules({NetworkModule.class})
 public class HistoryActivityTest {
-    @Rule //launches a given activity before the test starts and closes after the test
-    public ActivityScenarioRule<HistoryActivity> activityRule = new ActivityScenarioRule(HistoryActivity.class);
+
+    private static LocalPictureDatabase localPicDb = new LocalPictureDatabase(ApplicationProvider.getApplicationContext());
+
+    private HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+    @Rule
+    public RuleChain testRule = RuleChain.outerRule(hiltRule)
+            .around(new ActivityScenarioRule<>(HistoryActivity.class));
+
+    @BindValue
+    public static NetworkService networkInfo = mock(NetworkInformation.class);
+
+    @BeforeClass
+    public static void beforeAll(){
+        when(networkInfo.isNetworkAvailable()).thenReturn(false);
+        Location loc = new Location("");
+        loc.setLatitude(10);
+        loc.setLongitude(15);
+        Bitmap bmp = BitmapFactory.decodeResource(ApplicationProvider.getApplicationContext().getResources(), R.raw.ladiag);
+        LocalPicture pic =  new LocalPicture("testPic",bmp, loc, loc, new HashMap<>());
+        localPicDb.storePictureAndMetadata(pic);
```

Comment:
This pic is never deleted from the local db, try doing it in the teardown method

Reply:
Done